internal/app/orchestrator/orchestrator.go

```diff
@@ -187,24 +187,24 @@ func (o *orchestrator) Fetch(context.Context, discovery.DiscoveryRequest) (*gcp.
 func (o *orchestrator) watchUpstream(
 	ctx context.Context,
 	aggregatedKey string,
-	responseChannel <-chan *upstream.Response,
+	responseChannel <-chan *discovery.DiscoveryResponse,
 	done <-chan bool,
 ) {
 	for {
 		select {
-		case x := <-responseChannel:
-			if x.Err != nil {
+		case x, more := <-responseChannel:
+			if !more {
 				// A problem occurred fetching the response upstream, log and
 				// return the most recent cached response, so that the
 				// downstream will reissue the discovery request.
-				o.logger.With("err", x.Err).With("key", aggregatedKey).Error(ctx, "upstream error")
```

Comment:
Did we get rid of the `With("err", x.Err)` because it's a duplicate? Should still log the key imo.

Reply:
There is no `x.Err` anymore since we just close the channel and do not send the error to caller.

Reply:
What about the key? I think we should still include it in the log message

Reply:
yeah,,key should be there..removing that was a mistake.

Reply:
fixed it now